### DISCUSSION TITLE
Custom rules for Controller?

### DISCUSSION THREAD

**Discussion by aksalsf** [react-hook-form/react-hook-form#8841] [Q&A]
I just figure out how to apply custom rules for <Controller ... /> component, because when I set custom error using setError, it doesn't appear when I didn't define it on rules. My options now is figure out how to set custom rules or just remove the rules and handle it outside using setError.

**Comment by Moshyfawn** [comment]
```jsx
<Controller
  rules={{
    validate: {
      custom: value => validateCustom(value),
      rule: value => validateRule(value)
    }
  }}
/>
```
